### PR TITLE
Api error interceptor could not parse some failed responses where there is no api error body

### PR DESCRIPTION
**Pull request by vhontar**

### CODE REVIEW

File: library/src/commonMain/kotlin/com/ioki/passenger/api/IokiService.kt [hunk 1]
```diff
@@ -882,25 +882,21 @@ private suspend fun mapApiError(
...
-
-        interceptors.forEach { interceptor ->
-            if (interceptor.intercept(apiErrors, code)) {
-                return Result.Error.Api.Intercepted(apiErrors, code)
-            }
+    val apiErrorBody = runCatching { failedResponse.body<ApiErrorBody>() }.getOrNull() ?: ApiErrorBody()
+    val code = failedResponse.status.value
+    // 502 happens when Google runs internal tests. Catching it here prevents it from being reported as an error
+    val apiErrors =
+        if (code == com.ioki.passenger.api.result.HttpStatusCode.BAD_GATEWAY_502) {
```
Comment (StefMa) [inline library/src/commonMain/kotlin/com/ioki/passenger/api/IokiService.kt:889]: We can import that

**Review by StefMa — CHANGES_REQUESTED** [review]
Hey Vlad,
can you explain thst change to me please?!
I don't get it. Even though backend doesn't send us ErrorBody (which should be their fault? 🤔), why do we want to create an EMPTY ErrorBody and send them to the interceptors? Probably no interceptor is interested in it, right?

**Comment by vhontar** [comment]
@StefMa 
Thank you for the question!

No, we still interested in empty body, because we need only status code from API. Unfortunately it crashes the application if we try to parse empty ApiErrorBody and it just stops execution. But we still need status code for some interceptors only.

Does it make sense? :)

File: library/src/commonMain/kotlin/com/ioki/passenger/api/IokiService.kt (hunk 1 above)
Comment (vhontar) [reply library/src/commonMain/kotlin/com/ioki/passenger/api/IokiService.kt:889]: Yep, I’ll fix in the next PR 🙏🏻

**Comment by StefMa** [comment]
Thanks for the explanation!
You say the app crashes, is that really the case? Because we had everything wrapped here in a try catch? Or was the problem that we don't reach the interceptors?

I also don't like thst we construct an empty ErrorBody. In higher level modules we might think that backend sends us an empty body.
Would it make sense to try to parse a nullable ErrorBody instead? And put a null ErrorBody to the interceptors? Would this make more sense? 🤔

**Comment by vlh-ioki** [comment]
No, the doesn't crash because of the try/catch, but parser itself crashes, because there is empty body ApiBodyError sent by backend. 

I'd assume that `.body<ApiBodyError?>` won't work here.

Screenshot 2024-09-13 at 13 01 14

**Comment by vlh-ioki** [comment]
We need unit tests here: https://github.com/ioki-mobility/kmp-passenger-api/issues/22

**Pull request merged**